### ISSUE TITLE
use logstash as new data ingestion strategy

### ISSUE THREAD

**Issue by frankleng** [rethinkdb/elasticsearch-river-rethinkdb#7]
Be nice to have a way to specify exactly what is exposed to ES.

**Comment by frankleng** [comment]
@deontologician I know you are busy with other things. just wondering if you have any plans in the next month or two to implement this

**Comment by deontologician** [comment]
It's next on my list. I'm currently working on some webui things, but
probably in a week or so.

**Comment by codeincarnate** [comment]
For us, this is currently a huge hang up.  When the river adds the mapping to Elastic, there's not currently a way (to my knowledge) to specify fields that shouldn't be tokenized and analyzed.  

This makes it impossible to filter against fields which contain things like e-mail addresses.  They'll be tokenized and so matching doesn't work correctly.

**Comment by deontologician** [comment]
I just put in the code review for my previous project, so I'll be picking
this up today

**Comment by frankleng** [comment]
@codeincarnate I think you can work around it 
```
curl -XPUT '127.0.0.1:9200/table' -d '
{
  "mappings": {
    "table": {
      "properties": {
        "email": {
          "type": "string",
          "index": "not_analyzed"
        }
      }
    }
  }
}
```
put that mapping in before you initialize the rethinkdb indexes via river.

**Comment by codeincarnate** [comment]
@frankleng I tried that, didn't appear to work, though I was extremely tired when doing it.  I'll give it another shot.

**Comment by frankleng** [comment]
@codeincarnate  cool. be sure to delete existing indices and the _river index before you put in the custom mapping.

**Comment by deontologician** [comment]
Hey guys, I just put the initial (very very beta) version of the logstash plugin up at https://github.com/rethinkdb/logstash-input-rethinkdb/tree/master . 

Currently, it needs a logstash filter to be able to pipe directly into ElasticSearch, so this is one piece of the puzzle, not the entire thing. That being said, I'd really appreciate any early feedback you can provide if you're interested in playing around with it.

**Comment by frankleng** [comment]
@deontologician Thanks a lot! I'll definitely test it out this week.
Altho it seems like the two logstash limitations are also present in the river. so no loss there.

**Comment by deontologician** [comment]
@frankleng that's true, they are also present in the river, but I made more of an effort to try reconnecting and to add backfilling behavior since there was no expectation that changefeeds would grow these abilities soon. Since they're on the roadmap now, there is a bit less functionality in the logstash plugin

**Comment by frankleng** [comment]
@deontologician 
Tried this and works fine. but as you mentioned it's blocked by the implementation of resumable changefeeds. So we'll stick to the river for now. unless I've missed something. 

Thanks!

**Comment by deontologician** [comment]
awesome, thanks for checking it out

**Comment by deontologician** [comment]
closing, all discussion about the logstash plugin should continue at https://github.com/rethinkdb/logstash-input-rethinkdb/issues
